File: src/gestorAplicacion/servicios/Cita.java
/* Autores: Diego Andres Gracia Granados, Daniel Giraldo Vanegas, Elian David Velandia Riveros, Juan Camilo Gutierrez Martinez
y Santiago Arboleda Acevedo */
package gestorAplicacion.servicios;

import gestorAplicacion.personas.Doctor;
import gestorAplicacion.personas.Paciente;

//Clase destinada a crear citas médicas
public class Cita extends Servicio {
    //Atributos
    private Doctor doctor;
    protected String fecha;

    //Constructor
    public Cita(Doctor doctor, String fecha, Paciente paciente){
        super(paciente);
        this.doctor = doctor;
        this.fecha = fecha;
    }

    //Métodos

    //Métodos que busca el estado de pago de una cita médica y lo cambia
    @Override
    public void validarPago(Paciente paciente, long idServicio) {
        for (Cita cita :
                paciente.getHistoriaClinica().getHistorialCitas()) {
            if (cita.getIdServicio() == idServicio)
                cita.setEstadoPago(true);
        }
    }

    // Metodo que imprime una descripcion del servicio
    @Override
    public String descripcionServicio() {
        return idServicio + " - Cita medica con " + doctor.getNombre() + " (" + fecha + ")";
    }


    //Getters y Setters

    public Doctor getDoctor() {
        return doctor;
    }

    public String mensaje(){
        return "del servicio cita médica!";
    }

    public void setDoctor(Doctor doctor) {
        this.doctor = doctor;
    }

    public String getFecha() {
        return fecha;
    }

    public void setFecha(String fecha) {
        this.fecha = fecha;
    }

}

File: Python/src/gestor_aplicacion/servicios/cita.py
# Autores: Diego Andres Gracia Granados, Daniel Giraldo Vanegas, Elian David Velandia Riveros, Juan Camilo Gutierrez Martinez
# y Santiago Arboleda Acevedo

from src.gestor_aplicacion.servicios.servicio import Servicio


# Clase destinada a crear citas médicas
class Cita(Servicio):

    # Atributos y constructor
    def __init__(self, paciente, doctor, fecha):
        super().__init__(paciente)
        self._doctor = doctor
        self._fecha = fecha

    # Metodos

    # Metodo que imprime una descripcion del servicio
    def descripcion_servicio(self):
        return f"{self._id_servicio} - Cita medica con {self._doctor.nombre} ({self._fecha}))"

    # Métodos que busca el estado de pago de una cita médica y lo cambia
    # Recibe un objeto de la clase Paciente y un entero
    def confirmar_pago(self, paciente, id_servicio):
        for cita in paciente.historia_clinica.historial_citas:
            if cita.id_servicio == id_servicio:
                cita.estado_pago = True
                break

    @property
    def doctor(self):
        return self._doctor

    @doctor.setter
    def doctor(self, value):
        self._doctor = value

    @property
    def fecha(self):
        return self._fecha

    @fecha.setter
    def fecha(self, value):
        self._fecha = value
